addons/website_sale/static/src/js/website_sale_recently_viewed.js

```diff
@@ -0,0 +1,186 @@
+
+    _onAddToCart: function (ev) {
+        var self = this;
+        var $card = $(ev.currentTarget).closest('.card');
+        this._rpc({
+            route: "/shop/cart/update_json",
+            params: {
+                product_id: $card.find('input[data-product-id]').data('product-id'),
+                add_qty: 1
+            },
+        }).then(function (data) {
+            var $q = $(".my_cart_quantity");
+            if (data.cart_quantity) {
+                $q.parents('li:first').removeClass('d-none');
+                $q.html(data.cart_quantity).hide().fadeIn(600);
+            }
+
+            var $navButton = wSaleUtils.getNavBarButton('.o_wsale_my_cart');
+            wSaleUtils.animateClone($navButton, $(ev.currentTarget).parents('.o-product-card'), 25, 40);
+
+            self._dp.add(self._fetch()).then(self._render.bind(self));
+
+            if ($('.oe_cart')) {
+                $(".js_cart_lines").first().before(data['website_sale.cart_lines']).end().remove();
+                $(".js_cart_summary").first().before(data['website_sale.short_cart_summary']).end().remove();
+            }
+        });
+    },
+});
+
+var productsRecentlyViewedUpdate = publicWidget.Widget.extend({
+    selector: '#product_details',
+    events: {
+        'change input.product_id[name="product_id"]': '_onProductChange',
+    },
+    debounceValue: 3000,
+
+    /**
+     * @constructor
+     */
+    init: function () {
+        this._super.apply(this, arguments);
+        this._onProductChange = _.debounce(this._onProductChange, this.debounceValue);
+    },
+
+    //--------------------------------------------------------------------------
+    // Private
+    //--------------------------------------------------------------------------
+
+    _updateProductView: function ($target) {
+        var productId = parseInt($target.val());
+        if (utils.get_cookie('product_id_' + productId)) {
+            return;
+        }
+        this._rpc({
+            route: '/shop/products/recently_viewed_update',
+            params: {
+                product_id: productId,
+            }
+        }).then(function (res) {
+            if (res) {
+                if (res.visitor_id) {
+                    utils.set_cookie('visitor_id', res.visitor_id);
+                }
+                utils.set_cookie('product_id_' + productId, '', 30 * 60);
+            }
+        });
+    },
+
+    //--------------------------------------------------------------------------
+    // Handlers
+    //--------------------------------------------------------------------------
+
+        /**
+     * @private
+     * @param {Event} ev
+     */
+    _onProductChange: function (ev) {
+        this._updateProductView($(ev.currentTarget));
+    },
+});
+
+publicWidget.registry.productsRecentlyViewedUpdate = productsRecentlyViewedUpdate;
+
+return {
+    productsRecentlyViewedUpdate: productsRecentlyViewedUpdate,
```

Comment:
Just add in the registry at definition and don't return it since it can be retrieved from the registry.